preprocessing/nextclade/src/loculus_preprocessing/config.py

```diff
@@ -78,26 +78,33 @@ def generate_argparse_from_dataclass(config_cls: type[Config]) -> argparse.Argum
 
 
 def get_config(config_file: str | None = None) -> Config:
-    # Config precedence: CLI args > ENV variables > config file > default
+    """
+    Config precedence: Direct function arg > CLI args > ENV variables > config file > default
 
+    args:
+        config_file: Path to YAML config file - only used by tests
+    """
+
+    # Set just log level this early from env, so we can debug log during config loading
     env_log_level = os.environ.get("PREPROCESSING_LOG_LEVEL")
     if env_log_level:
         logging.basicConfig(level=env_log_level)
 
     parser = generate_argparse_from_dataclass(Config)
     args = parser.parse_args()
 
-    # Load default config
     config = Config()
-
-    # Overwrite config with config in config_file
-    if config_file:
-        config = load_config_from_yaml(config_file, config)
-    if args.config_file:
-        config = load_config_from_yaml(args.config_file, config)
-    if not config.backend_host:  # Check if backend_host wasn't set during initialization
+    if not args.backend_host:  # Set here so we can use organism
```

Comment:
Why have you changed from config.backend_host to args.backend_host here? What would happen if you passed backend_host in the config_file and not in CLI arguments? I think then it would also be overwritten?

Reply:
Yeah my bad